js/data-parsing/map_parsing.js

```diff
@@ -0,0 +1,58 @@
+// Function to parse the underlying maps in each exchange
+function mapParse(exchangeData, exchangeType) {
+  var highbids = exchangeData.highbids;
+  var lowasks = exchangeData.lowasks;
+
+  // Initializing temporary variables for sorting through the maps
+  var tmp_highbid = 0,
+  tmp_lowask = 1000000000000000,
+  tmp_amt_highbid = 0,
+  tmp_amt_lowask = 0;
+
+  // Iterating through each entry in the map to find highest value
+  highbids.forEach(function(value, key) {
+    // GDAX or Bit
+    if (exchangeType === "gdax" || exchangeType === "bitfinex") {
+      if (value.rate > tmp_highbid) {
+        tmp_highbid = value.rate;
+        tmp_amt_highbid = parseFloat(value.amount);
```

Comment:
On the subject of `parseFloat`, we should really do this conversion on the beginning of things as opposed to when we read the result. In data parsing, these amounts should be stored as numbers, if only for minimizing used memory since numbers are easier to store than strings of numbers. **O P T I M I Z A T I O N**